src/nnet3/nnet-general-component.h

```diff
@@ -775,7 +775,149 @@ class DropoutMaskComponent: public RandomComponent {
   = (const DropoutMaskComponent &other); // Disallow.
 };
 
+/*
+  Class SelfAttentionComponent is used together with
+  StatisticsExtractionComponent to extract weighted mean and
+  standard-deviation statistics.
+
+  SelfAttentionComponent is a part of the self-attention mechanism, which does 
+  a linear combination of a series of input frames
+
+ # In SelfAttentionComponent, the first n columns of the input matrix are interpreted
+ # as the weight vectors in multi-head attentions. If only a single-head attention is used,
+ # then the first column of the input matrix is the weight vector. The n + 1 th column of
+ # the input matrix is the count from the extraction component, it will not be used in 
```

Comment:
Does it mean that you always expect the previous layer to be StatisticExtraction component?